model/Tool/Factory/Lti1p3DeliveryLaunchCommandFactory.php

```diff
@@ -86,4 +89,12 @@ private function getLisOutcomeServiceUrlFactory(): LisOutcomeServiceUrlFactory
     {
         return $this->getServiceLocator()->get(LisOutcomeServiceUrlFactory::class);
     }
+
+    private function getReturnUrl()
+    {
+        if ($this->getServiceLocator()->has(ReturnUrlService::SERVICE_ID)) {
+            return $this->getServiceLocator()->get(ReturnUrlService::SERVICE_ID)->getReturnUrl();
+        }
+        return _url('index', 'DeliveryServer', 'taoDelivery');
```

Comment:
This is pretty much useless in case `ReturnUrlService` is not there, is it?
I mean, that would've essentially meant `taoDelivery` wasn't installed.

I suggest you fall back to `\tao_helpers_Uri::getRootUrl()` in this highly implausible case.

Reply:
fixed